api/v1alpha7/openstackcluster_types.go

```diff
@@ -155,13 +155,18 @@ type OpenStackClusterSpec struct {
 type OpenStackClusterStatus struct {
 	Ready bool `json:"ready"`
 
-	// Network contains all information about the created OpenStack Network.
-	// It includes Subnets and Router.
-	Network *Network `json:"network,omitempty"`
+	// Network contains information about the created OpenStack Network.
+	Network *NetworkStatusWithSubnets `json:"network,omitempty"`
 
 	// externalNetwork contains information about the external network used for default ingress and egress traffic.
```

Comment:
Not really part of your PR, would you mind changing that too?
```suggestion
	// ExternalNetwork contains information about the external network used for default ingress and egress traffic.
```

Reply:
I think we need a general godoc cleanup, tbh.